wallet/core/src/storage/local/collection.rs

```diff
@@ -110,13 +114,12 @@ where
     }
 
     pub fn load_multiple(&self, ids: &[Id]) -> Result<Vec<Arc<Data>>> {
-        Ok(ids
-            .iter()
+        ids.iter()
             .map(|id| match self.map.get(id).cloned() {
-                Some(data) => data,
-                None => panic!("requested id `{}` was not found in collection", id),
+                Some(data) => Ok(data),
+                None => Err(Error::KeyId(id.to_string())),
             })
-            .collect())
+            .collect::<Result<Vec<_>>>()
```

Comment:
I explicitly want this to fail to prevent wallet getting into any weird state. This denotes data integrity and if you are querying for missing data something is very wrong. OTOH you could resolve the error when you are querying but then this ships such checks upwards resulting in checks everywhere. 

I originally had it like you are suggesting and converted it to function in this manner explicitly.  (nite that it was a panic before, but that was too severe).